api/lib/subscription/slashing.js

```diff
@@ -38,20 +53,58 @@ class SlashingMonitor {
     )
 
     // requires some more logic to not spam the notifications if a validator is down for 1000 blocks
-    // this.client.subscribe(
-    //   { query: 'liveness.missed_blocks >= 1' },
-    //   (response) => {
-    //     const missedBlocks = response.events['liveness.address'].map(
-    //       (address, index) => ({
-    //         operatorAddress: address,
-    //         missedBlocks: response.events['liveness.missed_blocks'][index],
-    //         height: response.height
-    //       })
-    //     )
-    //     console.log('Missed block', missedBlocks)
-    //   }
-    // )
+    this.client.subscribe(
+      { query: `tm.event='NewBlockHeader' AND liveness.missed_blocks >= 1` },
+      async (response) => {
+        try {
+          const missedBlocks = response.events['liveness.address'].map(
+            (address, index) => ({
+              networkId: this.networkId,
+              operatorAddress: address,
+              blocks: response.events['liveness.missed_blocks'][index],
+              reason: 'missed_blocks',
+              height: response.data.value.header.height,
+            })
+          )
+          const rows = await Promise.all(missedBlocks.map(async missedBlockEvent => {
+            const lastMissedBlockEvent = await getLastMissedBlock(this.networkId, missedBlockEvent.operatorAddress)
+            return aggregateMissedBlocks(lastMissedBlockEvent, missedBlockEvent)
+          }))
+          database(config)("").upsert("slashes", rows)
+          console.log('Wrote missed block events', rows.length)
+        } catch (error) {
+          console.error("Failed to write missed block events", error)
+          Sentry.captureException(error)
+        }
```

Comment:
Probably this is an event that validators will be highly interested in (at least the rookie ones that are not measuring this themselves yet). Will we have in the future notifications specific for validators and others specific for "regular" users?

Reply:
That is the idea. Validators would get an event on the first missed block and users once a validator actually gets slashed or similar.

Reply:
Nice

Reply:
> Will we have in the future notifications specific for validators and others specific for "regular" users?

probably, yes.